dashboard/app/controllers/activities_controller.rb

```diff
@@ -124,6 +125,16 @@ def milestone
       is_ai_experiment_enabled = current_user && Experiment.enabled?(user: current_user, script: @script_level&.script, experiment_name: 'ai-rubrics')
       is_level_ai_enabled = EvaluateRubricJob.ai_enabled?(@script_level)
       if is_ai_experiment_enabled && is_level_ai_enabled && params[:submitted] == 'true'
+        metadata = {
+          'userId' => current_user.id,
+          'levelId' => @level.id,
+          'levelName' => @level.name,
```

Comment:
this metadata will need to match up across event types, and will need to uniquely identify the level/lesson, keeping mind mind that levels are shared across ai-enabled units. I would recommend logging script name and level name. 

on the client, script name and level name appear to already be available inside `reportingData`.

I'm not sure exactly how this will work in statsig yet, but to minimize the chances of us running into problems, please use the exact same names for these fields across event types, e.g. "studentId" here if that's what we're using in other places.

Reply:
Updated this